src/cpp/security/accesscontrol/DistinguishedName.h

```diff
@@ -0,0 +1,201 @@
+ * AttributeTypeAndValue
+ * AttributeTypeAndValue ::= SEQUENCE {
+ * type  AttributeType,
+ * value AttributeValue }
+ *
+ * Each RDN in a RDNSequence is separated by ','.
+ * Each AttributeTypeAndValue in a RDN is separated by '+'.
+ * Attribute and Value are separated by '='.
+ *
+ */
+
+/**
+ * @brief class that represents a Distinguished Name
+ *
+ * @note so far this implementation works only as a string
+ * @todo make this a real class with real methods that correctly implement RFC 2253:
+ */
+using DistinguishedName = std::string;
+
+/**
+ * @brief This function compares 2 strings representing DistinguishedNames.
+ *
+ * Follow RFC 2253: https://datatracker.ietf.org/doc/html/rfc2253
+ * The idea here is that comparing 2 DistinguishedNames is not as trivial as comparing 2 strings,
+ * nor comparing its values splitting the string.
+ * The comparison must be a complex function that takes into account many specific details.
+ *
+ * So far, Fast DDS gives 2 strings (in some format that we assume is the correct one) and this compares
+ * whether the two strings refer to the same DistinguishedName.
+ * It would be better (and easier) to store DistinguishedName in a specialized function instead of in a string,
+ * and thus this comparison would be much easier and efficient.
+ * But this will be left for future improvements.
+ *
+ * @warning this function does not correctly fulfilled all requirements of RFC 2253.
+ * Requirements that are not fulfilled:
+ * - multi-attribute
+ * - types accept oid
+ * - assumes that input is a well formed DistinguishedName following rfc2253
+ * @todo make this function complete (probably better by implementing DistinguishedName class).
+ *
+ * @param name1 first DistinguishedName in string format
+ * @param name2 second DistinguishedName in string format
+ *
+ * @return true if both first strings represent the same DistinguishedName.
+ * @return false otherwise
+ */
+bool rfc2253_string_compare(
+        const DistinguishedName& name1,
+        const DistinguishedName& name2);
+
+
+namespace details {
+
+struct Attribute
+{
+    Attribute() = default;
+
+    Attribute(
+            const DistinguishedName& name);
+
+    Attribute(
+            const char* name);
+
+    bool is_set() const noexcept;
+
+    size_t cut(
+            size_t ini,
+            size_t fin = 0) noexcept;
+
+    size_t trim_blank_spaces();
+
+    size_t trim_back_blank_spaces();
+
+    size_t clear() noexcept;
+
+    size_t find(
+            char c) const noexcept;
+
+    size_t find(
+            char c,
+            bool& result) const noexcept;
+
+    bool found(
+            const size_t& find_result) const noexcept;
+
+    bool operator ==(
+            const Attribute& other) const noexcept;
+
+    bool operator !=(
+            const Attribute& other) const noexcept;
+
+    static Attribute cut(
+            const Attribute& att,
+            size_t ini,
+            size_t fin = 0) noexcept;
+
```

Comment:
As a suggestion, although inefficient but it could be helpful adding a operator `<<` overload. Something like
```suggestion
    friend std::ostream& operator<<(std::ostream& os, const Attribute& att)
    {
        size_t size = att.length;
        const char *ch = att.value;
        while (size)
        {
            os << *ch;
            ++ch;
            --size;
        }

        return os;
    }
```

Reply:
I though about it but didn't do it because it was not used. I add it (with a slight modification)